sdk/src/wallet/core/builder.rs

```diff
@@ -247,6 +247,30 @@ where
             self.alias = loaded_wallet_builder.as_ref().and_then(|builder| builder.alias.clone());
         }
 
+        // May use a previously stored wallet address if it wasn't provided
+        if self.address.is_none() {
+            self.address = loaded_wallet_builder
+                .as_ref()
+                .and_then(|builder| builder.address.clone());
+        }
+
+        // Create the node client.
+        let client = self
+            .client_options
+            .clone()
+            .ok_or(crate::wallet::Error::MissingParameter("client_options"))?
+            .finish()
+            .await?;
+
+        // May create a default Ed25519 wallet address if there's a secret manager.
+        let wallet_address = if self.address.is_none() && self.secret_manager.is_some() {
+            let bech32_hrp = client.network_info.read().await.protocol_parameters.bech32_hrp();
+            let address = self.create_default_wallet_address(bech32_hrp).await?;
+            address
+        } else {
+            return Err(crate::wallet::Error::MissingParameter("address"));
+        };
```

Comment:
This all looks like the result of a failed conflict resolution?

Reply:
I re-added it because it didn't make sense to me that we can't generate an address now

Reply:
But `self.address` can't be None at this point

Reply:
Why not? If you provide the bip path it should generate it in L237?

Reply:
oh, missed that haha